packages/hoppscotch-sh-admin/src/composables/usePagedQuery.ts

```diff
@@ -16,38 +16,41 @@ export function usePagedQuery<
   const { client } = useClientHandle();
   const fetching = ref(true);
   const error = ref(false);
-  const list = ref<ListItem[]>([]);
+  const list: Ref<ListItem[]> = ref([]);
   const currentPage = ref(0);
   const hasNextPage = ref(true);
 
   const fetchNextPage = async () => {
     fetching.value = true;
 
-    try {
-      const cursor =
-        list.value.length > 0 ? getCursor(list.value.at(-1)) : undefined;
-      const variablesForPagination = {
-        ...variables,
-        take: itemsPerPage,
-        cursor,
-      };
+    const cursor =
+      list.value.length > 0 ? getCursor(list.value.at(-1)!) : undefined;
```

Comment:
If the list has elements, `list.value.at(-1)` is guaranteed to have a last item.